src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java

```diff
@@ -0,0 +1,318 @@
+   */
+  private static boolean isDependencyCheck(Plugin plugin) {
+    return isDependencyCheck(plugin.getGroupId(), plugin.getArtifactId());
+  }
+
+  /**
+   * Check if a report plugin is OWASP Dependency Check plugin.
+   *
+   * @param plugin The report plugin to be checked.
+   * @return True if the report plugin is OWASP Dependency Check plugin, false otherwise.
+   */
+  private static boolean isDependencyCheck(ReportPlugin plugin) {
+    return isDependencyCheck(plugin.getGroupId(), plugin.getArtifactId());
+  }
+
+  /**
+   * Check if groupId and artifactId matches with OWASP Dependency GA.
+   *
+   * @param groupId to be checked.
+   * @param artifactId to be checked.
+   * @return True if the OWASP Dependency GA matches, false otherwise.
+   */
+  private static boolean isDependencyCheck(String groupId, String artifactId) {
+    return "org.owasp".equals(groupId) && "dependency-check-maven".equals(artifactId);
+  }
+
+  /**
+   * Check the availability of OWASP Dependency plugin in the project.
+   * 
+   * @param locations Set of location hits which indicates where exactly the plugin was found.
+   * @return The availability status of type {@link Status}.
+   */
+  private static Status availablityFrom(Set<Location> locations) {
+    if (locations == null || locations.isEmpty()) {
+      return NOTFOUND;
+    }
+
+    if (mandatoryCheckFor(locations)) {
+      return MANDATORY;
+    }
+
+    return OPTIONAL;
+  }
+
+  /**
+   * Check if plugin configurations have an attribute which can fail the build on finding
+   * vulnerabilities above a given cvss score. Hence, look for possible cvss score value which
+   * indicates, when to fail the build. The cvss score is in between 0.0 <= x <= 10.0. If the 
+   * score is > 11.0, then no vulnerabilities are considered to fail the build as this value 
+   * is out of bound.
+   *
+   * @param configuration The plugin configuration to be checked.
+   * @return The cvss score which indicates how the build should fail.
+   */
+  private static double cvssScoreFrom(Object configuration) {
+    // The OWASP dependency plugin is found and checking configurations.
+    for (String config : FAIL_BUILD_CONFIGURATIONS.keySet()) {
+      Optional<String> content = domParser(config, configuration);
+      String type = FAIL_BUILD_CONFIGURATIONS.get(config);
+
+      if (content.isPresent()) {
+        return parseScore(content.get(), type);
+      }
+    }
+
+    return OUT_OF_BOUND_CVSS_SCORE;
+  }
+
+  /**
+   * Parse the value depending on the given type. The type can be 'boolean' or 'number'. If
+   * 'boolean', then return 0.0, otherwise parse the given string value to 'number'.
+   * 
+   * @param value the value to be parsed.
+   * @param type the type of the string value.
+   * @return the parsed value.
+   */
+  private static double parseScore(String value, String type) {
+    if ("number".equals(type)) {
+      return Double.parseDouble(value);
+    }
+    return 0.0;
```

Comment:
To prevent potential issues, it would be better to return 0 if the the type is `boolean` and throw an exception in case of an unexpected type.